mne_bids/tests/test_write.py

```diff
@@ -511,14 +524,15 @@ def test_fif(_bids_validate):
     raw = _read_raw_fif(raw_fname)
     raw._filenames = (raw.filenames[0].replace('.fif', '.foo'),)
     with pytest.raises(ValueError, match='Unrecognized file format'):
-        write_raw_bids(raw, bids_path, bids_root)
+        write_raw_bids(raw, bids_path)
 
 
 @pytest.mark.skipif(LooseVersion(mne.__version__) < LooseVersion('0.20'),
                     reason="requires mne 0.20.dev0 or higher")
 def test_fif_anonymize(_bids_validate):
     """Test write_raw_bids() with anonymization fif."""
     bids_root = _TempDir()
+    bids_path.update(root=bids_root)
```

Comment:
same here. Where is bids_path defined? this can have side effects?

Reply:
It's defined in the global scope. I've now changed the tests to start off with producing a copy of that variable before proceeding to change it, see da67382

This should be turned into a fixture at one point.

Also I'm generally unhappy with the tests (and have been for a long time) bc they still may have sh*tloads of side effects we're not even aware of, the test functions are just way to big and constantly re-using stuff that we're fiddling with…